src/components/Table/components/CollapseToggle.tsx

```diff
@@ -21,11 +23,16 @@ export function CollapseToggle(props: GridTableCollapseToggleProps) {
     return null;
   }
 
+  const onClickToggleCollapsed = () => tableState.toggleCollapsed(row.id);
+
   return (
-    <IconButton
-      onClick={() => tableState.toggleCollapsed(row.id)}
-      icon={isHeader ? headerIconKey : iconKey}
-      compact={compact}
-    />
+    <>
+      {label ? (
+        <span css={Css.cursorPointer.$} onClick={onClickToggleCollapsed}>
+          {label}
+        </span>
+      ) : null}
+      <IconButton onClick={onClickToggleCollapsed} icon={isHeader ? headerIconKey : iconKey} compact={compact} />
```

Comment:
**question:** Where is the request for this label coming from? I'm curious to see the mocks that are changing our default behaviors for the table layouts.

Reply:
Hi @bdow, sure, this design was developed in the new Creation Invoice workflow, you can see more details in the Figma by this [URL](https://www.figma.com/file/Iwg2Dse4GL5pBaBuiAUZBo/Q3-2022-%7C-Cost-Plus-Invoicing?node-id=933%3A3497&t=RGmgdqJVnEpThha5-0) or this [SC Epic](https://app.shortcut.com/homebound-team/epic/21011/cost-plus-upgrade-cpu?vc_group_by=day&cf_workflow=500000021&ct_workflow=all&workflow_state_ids=500000008&workflow_state_ids=500000007).

Reply:
I've commented on the Figma to ask the designer if we really want to break convention for this workflow. Let's wait to hear what they say before moving forward with this.
FWIW, I'm thinking this is kind of a one-off pattern that we might not want to define in Beam. If we see it pop up in more places, then sure. But for now I think it might be worth creating a similar "CollapseRowToggle" component directly in Blueprint that does almost this same thing.
I also think it'd probably do something more like:
```
return <Button variant="text" label={label} onClick={onClickToggleCollapsed} endAdornment={<Icon icon={iconKey} />} />
```

Reply:
I understood @bdow, so I created a new PR in BP to create the proposed component. Can you take a look please?

https://github.com/homebound-team/internal-frontend/pull/3112